src/defrag.c

```diff
@@ -149,15 +151,13 @@ DefragFragInit(void *data, void *initdata)
 void
 DefragTrackerFreeFrags(DefragTracker *tracker)
 {
-    Frag *frag;
+    Frag *frag, *tmp;
 
     /* Lock the frag pool as we'll be return items to it. */
     SCMutexLock(&defrag_context->frag_pool_lock);
 
-    while ((frag = TAILQ_FIRST(&tracker->frags)) != NULL) {
-        TAILQ_REMOVE(&tracker->frags, frag, next);
-
-        /* Don't SCFree the frag, just give it back to its pool. */
+    RB_FOREACH_SAFE(frag, IP_FRAGMENTS, &tracker->fragment_tree, tmp) {
+        RB_REMOVE(IP_FRAGMENTS, &tracker->fragment_tree, frag);
         DefragFragReset(frag);
         PoolReturn(defrag_context->frag_pool, frag);
```

Comment:
PoolGet is wrapped in a pool lock, but the return isn't. Is that correct?

Reply:
This return is wrapped in a lock. The other return in DefragInsert isn't though. The lock could probably be just around PoolReturn instead of the whole loop though.

Reply:
I think this one is probably fine? I commented on the wrong line.

Reply:
Yeah, my error. Fixed in next PR.